src/PrestaShopBundle/Resources/config/routing/admin/configure/advanced_parameters/security.yml

```
@@ -0,0 +1,83 @@
+admin_security:
+    path: /
+    methods: [GET]
+    defaults:
+        _controller: 'PrestaShopBundle:Admin\Configure\AdvancedParameters\Security:index'
+        _legacy_controller: AdminSecurity
+        _legacy_link: AdminSecurity
+
+admin_security_general_save:
+    path: /general
+    methods: [POST]
+    defaults:
+        _controller: 'PrestaShopBundle:Admin\Configure\AdvancedParameters\Security:processGeneralForm'
+        _legacy_controller: AdminSecurity
+        _legacy_link: AdminSecurity
+
+admin_security_sessions_customer_list:
+    path: /session/customer
+    methods: [GET]
+    defaults:
+        _controller: 'PrestaShopBundle:Admin\Configure\AdvancedParameters\Security:customerSession'
+        _legacy_controller: AdminSecuritySessionCustomer
+        _legacy_link: AdminSecuritySessionCustomer
+
+admin_security_sessions_customer_delete:
+    path: /session/customer/{sessionId}
```

Path seems to be wrong, currently it seems as if the resource you are targeting would be the customer and not the session. In theory It should be smth like
`/customers/sessions/{sessionId}` and customers`/employees/sessions/{sessionId}` () maybe? :thinking: (but then it conflicts with the Customer & Employees routes, idk if thats alright)
This is abit specific case, I'm not sure :sweat: . 
@matks, @jolelievre wdyt?

---

I'm in a `/security` context, then I'm looking for `session` which is dedicated to a specific customer or an employee.
With your idea, it should be `/customers/security/sessions` or `/customers/sessions/security` :thinking:

---

@PierreRambaud If we use the API, then when we fetch an order detail, I think it must be something like
`/orders/order_details/{id}` . And we retrieve a order_detail object.

I think indeed a lot of people are used to URLs as "the left part is the group, the domain, while the right part is what I am working with".

See also
- `PATCH /orders/update/{id}` => I Update an Order by its ID
-  `GET /customers/address/{id}` => I fetch a customer address by its ID

So I understand @zuk3975 confusion 😅  .

May I suggest to simply remove any confusion by being over-explicit?

```suggestion
    path: /session/delete-by-customer/{sessionId}
```
If we make it clear what is the object we want to delete what is the object we want to use as selector, then we have no confusion

---

It's against all HTTP best practices.
The HTTP method is here to define what kind of action should be executed :thinking: 

Don't forget all good API are using the same route, but with different HTTP method.

I really dislike this kind of HTTP request (`delete-by-something`) inside the URL, it's quite ugly :( , you should think about context.
I want to manage `security session` from `customers` or `employees`, think about customers and employees as a filter or a specific resource inside this context.
Otherwise, it should be in `Sell -> Customers` and the other in `Adavanced parameters -> Team`.

---

As I've said in slack, I would expect either generalized `sessions/{sessionId}` or dedicated `employee-sessions/{employeeSessionId} & customer-sessions/{customerSessionId}`.
From php part I see you have EmployeeSessionId & CustomerSessionId that suggests the second option. But that's just my opinion :man_shrugging: